src/models/room.ts

```diff
@@ -2197,7 +2197,7 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         // Due to replies not being typical relations and being used as fallbacks for threads relations
         // If we bypass the if case above then we know we are not a thread, so if we are still a reply
         // then we know that we must be in the main timeline. Same goes if we have no associated parent event.
-        if (!parentEventId || !!event.replyEventId) {
+        if (!parentEventId || !!event.replyEventId || event.getRelation()?.rel_type === "m.reference") {
```

Comment:
I'm not sure I really see why an `m.reference` relation needs special treatment here: doesn't the logic apply to any relation that's not a thread (or possibly reply?) relation? Seems like any event with a kind of relation we don't know about, we should assume lives in the main thread? That said, I'd appreciate a second pair of eyes on this before actually suggesting this.

Also, the comment above (apart from having half a sentence) explains both clauses in the if condition, so if you add another, please update the comment to reflect it.

Reply:
At the moment I'm more inclined to go with the other PR, which seems broadly sensible. The more I think about this, the more I think that actually there's no way we can legitimately say events live in any timeline if we don't know about their parent, so what we should actually be doing is keeping them in a holding area somewhere until we can fetch their parent. That or use https://github.com/matrix-org/matrix-spec-proposals/pull/4023